backends/hip/hip-vector.hip

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017, Lawrence Livermore National Security, LLC. Produced at
```

Comment:
We perhaps want to prepend `ceed-` on this file (and it's CUDA counterpart) to match the general style we use in the library

Reply:
I think the convention was that device kernels and files other than the ones specifically interacting with the libCEED API would have names without `ceed-` at the front to differentiate.  For `hip-ref` and `cuda-ref`, that's only the vector file; in the magma backend, we have a lot of device files, and they are all just named by whatever kernel/magma-related thing is in the file.  
So, I guess I would have a slight preference for keeping the names as-is, but I don't mind changing them.

Reply:
The CUDA backend looks like its 50-50 with it's .cu files. It looks like the new OCCA backend puts all of the kernels in a `kernels` folder, which might be the best approach? Not that it makes a big difference, but similarity between backends is nice.

Reply:
Wait, what am I missing?  I don't see any *.cu files that start with `ceed-`?
Moving everything to `kernels` might be a tidy solution though.

Reply:
its `/backend/cuda-reg/cuda-reg-basis.cu`

Reply:
Right, but that's `cuda-reg-basis.cu`, not `ceed-reg-basis.cu`
Unless you mean having the backend directory name as the beginning of the file, in which case magma is the odd one out.

Reply:
You're right. I just can't read today.